mlir/lib/Gradient/Transforms/GradMethods/PS_QuantumGradient.cpp

```diff
@@ -299,7 +299,13 @@ func::FuncOp ParameterShiftLowering::genQGradFunction(PatternRewriter &rewriter,
             else if (auto gate = dyn_cast<quantum::QuantumGate>(op)) {
                 // We are undoing the def-use chains of this gate's return values
                 // so that we can safely delete it (all quantum ops must be eliminated).
-                rewriter.replaceOp(gate, gate.getQubitOperands());
+                /* rewriter.replaceOp(gate, gate.getQubitOperands()); */
+                SmallVector<Value> values;
+                values.insert(values.end(), gate.getQubitOperands().begin(),
+                              gate.getQubitOperands().end());
+                values.insert(values.end(), gate.getCtrlQubitOperands().begin(),
+                              gate.getCtrlQubitOperands().end());
+                rewriter.replaceOp(gate, values);
```

Comment:
Same here, with a good interface this shouldn't be required.

Reply:
Indeed! Done